File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for (const string& token : tokens) {
            if (token.size() == 1 && (token[0] == '+' || token[0] == '-' || token[0] == '*' || token[0] == '/')) {
                int b = st.top(); st.pop();
                int a = st.top(); st.pop();
                if (token[0] == '+') st.push(a + b);
                else if (token[0] == '-') st.push(a - b);
                else if (token[0] == '*') st.push(a * b);
                else st.push(a / b);
            } else {
                st.push(stoi(token));
            }
        }
        return st.top();
    }
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for i in tokens:
            if i in {"+", "-", "*", "/"}:
                a = st.pop()
                b = st.pop()
                if i == '+':
                    st.append(b + a)
                elif i == '-':
                    st.append(b - a)
                elif i == '*':
                    st.append(b * a)
                else:
                    st.append(int(b / a))
            else:
                st.append(int(i))
        return st[-1]